### PR TITLE
Update bins slicing logic

### PR DESCRIPTION
**Pull request by YooSunYoung**
Update bins slicing logic to adjust an undefined start/stop when too small/big stop/start given.
Fixes https://github.com/scipp/scipp/issues/2874

### CODE REVIEW

File: tests/slice_bins_test.py [hunk 1]
```diff
@@ -59,23 +59,32 @@ def test_slice_bins_by_open_range_includes_everything():
     assert result.bins.size().sum().value == 100
 
 
-def test_slice_bins_by_half_open_int_range_splits_without_duplication():
+def test_slice_bins_by_half_open_float_range_splits_without_duplication():
```
Comment (YooSunYoung) [inline tests/slice_bins_test.py:62]: I changed the names since the splitting range seemed float in this case. Please correct me if I was wrong...!

**Comment by YooSunYoung** [comment]
> See my review in #2932, not sure why it got closed.

Sorry I just renamed the branch...! I thought it will be fixed in the pull request automatically but apparently not....

**Review by SimonHeybrock — COMMENTED** [review]
Looks good!

One more small thing to do: For most changes, we briefly mention them in them in the release notes (and link to the PR). See https://github.com/scipp/scipp/blob/main/docs/about/release-notes.rst.

**Comment by YooSunYoung** [comment]
> 

I updated the release note. Can you please have a look? @SimonHeybrock

**Pull request merged**